app/jobs/add_registration_job.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class AddRegistrationJob < ApplicationJob
+  def perform(lane, competition_id, user_id, lane_params)
+    ActiveRecord::Base.transaction do
+      if lane == "competing"
+        registration = Registration.build(competition_id: competition_id, user_id: user_id)
+        Registrations::CompetingLane.create!(registration, lane_params[:competing], user_id, competition_id)
```

Comment:
My suggestion: Use the `lane` parameter to perform some kind of (very rudimentary) lookup to find the module/class that handles this lane, and then give the lane a method along the lines of `process`.

The implementation can then do whatever the heck it wants with the raw payload. In this case, the body of that method over in the `lib` code should be the one to create the registration entity.

Because strictly speaking, the `Registration` model in the legacy backend is (mostly, except for the `is_competing` flag which was hacked in waaaaay later as a low-effort quick and easy approach for Continental and World staff planning) actually a `competing_registration` model. So it would make sense imo if it lives under the auspices of the competing lane